Full application crash at startup on version 0.8.7 (F-droid)

**Summary:** 

Application crash on startup

**Steps to reproduce:** 

Install the application from F-droid, launch it, it crashes.

**Add system logs:**

`adb logcat -v threadtime` result:
```
04-08 16:03:41.902  3844  3844 I Timeline: Timeline: Activity_launch_request id:openfoodfacts.github.scrachx.openfood time:677415936
04-08 16:03:41.903   927  3249 I ActivityManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=openfoodfacts.github.scrachx.openfood/.views.splash.SplashActivity bnds=[32,203][196,393]} from uid 10110 on display 0
04-08 16:03:41.954   927  3026 V BoostFramework: BoostFramework() : mPerf = com.qualcomm.qti.Performance@15c70dd
04-08 16:03:41.954   927  3026 I ActivityManager: Start proc 18456:openfoodfacts.github.scrachx.openfood/u0a366 for activity openfoodfacts.github.scrachx.openfood/.views.splash.SplashActivity
04-08 16:03:42.003  3844  4170 D OpenGLRenderer: endAllActiveAnimators on 0x9b657100 (StateListDrawable) with handle 0x9f92a380
04-08 16:03:42.019 18456 18456 W System  : ClassLoader referenced unknown path: /data/app/openfoodfacts.github.scrachx.openfood-1/lib/arm
04-08 16:03:42.023 18456 18456 I MultiDex: VM with version 2.1.0 has multidex support
04-08 16:03:42.023 18456 18456 I MultiDex: Installing application
04-08 16:03:42.023 18456 18456 I MultiDex: VM has multidex support, MultiDex support library is disabled.
04-08 16:03:42.033 18456 18456 I art     : Rejecting re-init on previously-failed class java.lang.Class<org.b.a.b.b$a>
04-08 16:03:42.033 18456 18456 I art     : Rejecting re-init on previously-failed class java.lang.Class<org.b.a.b.b$a>
04-08 16:03:42.033 18456 18456 I art     : Rejecting re-init on previously-failed class java.lang.Class<org.b.a.b.b$a>
04-08 16:03:42.033 18456 18456 I art     : Rejecting re-init on previously-failed class java.lang.Class<org.b.a.b.b$a>
04-08 16:03:42.033 18456 18456 I art     : Rejecting re-init on previously-failed class java.lang.Class<org.b.a.b.b$a>
04-08 16:03:42.033 18456 18456 I art     : Rejecting re-init on previously-failed class java.lang.Class<org.b.a.b.b$a>
04-08 16:03:42.033 18456 18456 I art     : Rejecting re-init on previously-failed class java.lang.Class<org.b.a.b.b$a>
04-08 16:03:42.034 18456 18456 I art     : Rejecting re-init on previously-failed class java.lang.Class<org.b.a.b.b$a>
04-08 16:03:42.034 18456 18456 I art     : Rejecting re-init on previously-failed class java.lang.Class<org.b.a.b.b$a>
04-08 16:03:42.034 18456 18456 I art     : Rejecting re-init on previously-failed class java.lang.Class<org.b.a.b.b$a>
04-08 16:03:42.034 18456 18456 I art     : Rejecting re-init on previously-failed class java.lang.Class<org.b.a.b.b$a>
04-08 16:03:42.035 18456 18456 I art     : Rejecting re-init on previously-failed class java.lang.Class<org.b.a.b.b$a>
04-08 16:03:42.035 18456 18456 I art     : Rejecting re-init on previously-failed class java.lang.Class<org.b.a.b.b$a>
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: Failed to instantiate custom view inflater android.support.v7.app.AppCompatViewInflater. Falling back to default.
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: java.lang.ClassNotFoundException: android.support.v7.app.AppCompatViewInflater
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at java.lang.Class.classForName(Native Method)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at java.lang.Class.forName(Class.java:324)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at java.lang.Class.forName(Class.java:285)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at android.support.v7.app.l.b(Unknown Source)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at android.support.v7.app.l.onCreateView(Unknown Source)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:746)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:704)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at android.view.LayoutInflater.inflate(LayoutInflater.java:492)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at android.view.LayoutInflater.inflate(LayoutInflater.java:423)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at android.view.LayoutInflater.inflate(LayoutInflater.java:374)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at com.android.internal.policy.PhoneWindow.generateLayout(PhoneWindow.java:3923)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at com.android.internal.policy.PhoneWindow.installDecor(PhoneWindow.java:3995)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at com.android.internal.policy.PhoneWindow.getDecorView(PhoneWindow.java:1976)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at c.a.a.a.<init>(Unknown Source)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at openfoodfacts.github.scrachx.openfood.views.splash.SplashActivity.onCreate(Unknown Source)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at android.app.Activity.performCreate(Activity.java:6285)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1108)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2417)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2524)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at android.app.ActivityThread.access$900(ActivityThread.java:154)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1391)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at android.os.Handler.dispatchMessage(Handler.java:102)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at android.os.Looper.loop(Looper.java:234)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at android.app.ActivityThread.main(ActivityThread.java:5526)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at java.lang.reflect.Method.invoke(Native Method)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: Caused by: java.lang.ClassNotFoundException: Didn't find class "android.support.v7.app.AppCompatViewInflater" on path: DexPathList[[zip file "/data/app/openfoodfacts.github.scrachx.openfood-1/base.apk"],nativeLibraryDirectories=[/data/app/openfoodfacts.github.scrachx.openfood-1/lib/arm, /vendor/lib, /system/lib]]
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at java.lang.ClassLoader.loadClass(ClassLoader.java:511)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	at java.lang.ClassLoader.loadClass(ClassLoader.java:469)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	... 27 more
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	Suppressed: java.lang.ClassNotFoundException: android.support.v7.app.AppCompatViewInflater
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 		at java.lang.Class.classForName(Native Method)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 		at java.lang.BootClassLoader.findClass(ClassLoader.java:781)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 		at java.lang.BootClassLoader.loadClass(ClassLoader.java:841)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 		at java.lang.ClassLoader.loadClass(ClassLoader.java:504)
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 		... 28 more
04-08 16:03:42.099 18456 18456 I AppCompatDelegate: 	Caused by: java.lang.NoClassDefFoundError: Class not found using the boot class loader; no stack trace available
04-08 16:03:42.108 18456 18456 W com.b.a.c.e.a: Unable to load JDK7 types (annotations, java.nio.file.Path): no Java7 support added
04-08 16:03:42.109 18456 18456 D AndroidRuntime: Shutting down VM
04-08 16:03:42.109 18456 18456 E AndroidRuntime: FATAL EXCEPTION: main
04-08 16:03:42.109 18456 18456 E AndroidRuntime: Process: openfoodfacts.github.scrachx.openfood, PID: 18456
04-08 16:03:42.109 18456 18456 E AndroidRuntime: java.lang.NoSuchFieldError
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at libcore.reflect.AnnotationAccess.decodeValue(AnnotationAccess.java:688)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at libcore.reflect.AnnotationAccess.toAnnotationInstance(AnnotationAccess.java:663)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at libcore.reflect.AnnotationAccess.toAnnotationInstance(AnnotationAccess.java:641)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at libcore.reflect.AnnotationAccess.getDeclaredAnnotation(AnnotationAccess.java:170)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at libcore.reflect.AnnotationAccess.getAnnotation(AnnotationAccess.java:72)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at java.lang.Class.getAnnotation(Class.java:359)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at com.b.a.c.f.y$a.<clinit>(Unknown Source)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at com.b.a.c.f.y$a.a(Unknown Source)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at com.b.a.c.s.<clinit>(Unknown Source)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at g.b.a.a.a(Unknown Source)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at openfoodfacts.github.scrachx.openfood.c.a.<init>(Unknown Source)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at openfoodfacts.github.scrachx.openfood.c.a.a(Unknown Source)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at openfoodfacts.github.scrachx.openfood.d.b.<init>(Unknown Source)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at openfoodfacts.github.scrachx.openfood.d.b.b(Unknown Source)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at openfoodfacts.github.scrachx.openfood.views.splash.b.<init>(Unknown Source)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at openfoodfacts.github.scrachx.openfood.views.splash.SplashActivity.onCreate(Unknown Source)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:6285)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1108)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2417)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2524)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at android.app.ActivityThread.access$900(ActivityThread.java:154)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1391)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:102)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:234)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:5526)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: Caused by: java.lang.NoSuchFieldException: No field PUBLIC_ONLY in class Lcom/b/a/a/e$a; (declaration of 'com.b.a.a.e$a' appears in /data/app/openfoodfacts.github.scrachx.openfood-1/base.apk)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at java.lang.Class.getDeclaredField(Native Method)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	at libcore.reflect.AnnotationAccess.decodeValue(AnnotationAccess.java:685)
04-08 16:03:42.109 18456 18456 E AndroidRuntime: 	... 27 more
04-08 16:03:42.111   927  3857 D ActivityManager: New dropbox entry: openfoodfacts.github.scrachx.openfood, data_app_crash, d8cea668-c3b4-4fef-b034-907eb6bc4207
04-08 16:03:42.112   927  3857 W ActivityManager:   Force finishing activity openfoodfacts.github.scrachx.openfood/.views.splash.SplashActivity
04-08 16:03:42.163 12597 18496 W ContextImpl: Calling a method in the system process without a qualified user: android.app.ContextImpl.bindService:1271 android.content.ContextWrapper.bindService:604 com.sonyericsson.crashmonitor.MiscTaAdapter.open:90 com.sonyericsson.crashmonitor.service.CrashMonitorService.onInit:127 com.sonyericsson.crashmonitor.service.CrashMonitorService.onHandleIntent:192
04-08 16:03:42.179   927   846 I OpenGLRenderer: Initialized EGL, version 1.4
04-08 16:03:42.633   927   955 W ActivityManager: Activity pause timeout for ActivityRecord{fdba2ee u0 openfoodfacts.github.scrachx.openfood/.views.splash.SplashActivity t5957 f}
04-08 16:03:42.698  3844  3844 I Timeline: Timeline: Activity_idle id: android.os.BinderProxy@474de05 time:677416732
04-08 16:03:42.705   927  4142 I ActivityManager: Killing 17013:com.sonyericsson.tvout.mhl/1000 (adj 15): empty #17
04-08 16:03:42.846   927   978 I Timeline: Timeline: Activity_windows_visible id: ActivityRecord{d83b8ee u0 com.sonyericsson.home/com.sonymobile.home.HomeActivity t5658} time:677416880
04-08 16:03:43.674  3192  3192 W RecentsTaskLoader: Missing ActivityInfo for ComponentInfo{com.ChillyRoom.DungeonShooter/com.soulgame.sgsdk.tgsdklib.unity.TGSDKUnityActivity} u=0
04-08 16:03:44.256 18456 18456 I Process : Sending signal. PID: 18456 SIG: 9
04-08 16:03:44.271   927  4145 W InputMethodManagerService: Window already focused, ignoring focus gain of: com.android.internal.view.IInputMethodClient$Stub$Proxy@d879d6 attribute=null, token = android.os.BinderProxy@44d3ac0
04-08 16:03:44.296   927  4142 I ActivityManager: Process openfoodfacts.github.scrachx.openfood (pid 18456) has died
04-08 16:03:44.454   927   978 W AppOps  : Finishing op nesting under-run: uid 1000 pkg android code 24 time=0 duration=0 nesting=0
```

**Expected behavior:** 

To not crash.

**Observed behavior:** 

It crashed at startup

**Device and Android version:** 

Sony Z3 Compact, Android 6.0.1 with manufacturer unmodified rom.

Issue should probably be renamed after finding out what the issue is.

Especially, logs seems to state it can't find `/data/app/openfoodfacts.github.scrachx.openfood-1/lib/arm` and to have issues finding `AppCompatViewInflater`

EDIT: This is a clean installed. Removed everything related to OpenFood Facts before and then reinstalled it.

--- a/app/proguard-rules.pro
+++ b/app/proguard-rules.pro
@@ -20,9 +20,10 @@
 -keep @com.bluelinelabs.logansquare.annotation.JsonObject class *
 -keep class **$$JsonObjectMapper { *; }
 -keep class openfoodfacts.github.scrachx.openfood.models.** { *; }
+-keep class openfoodfacts.github.scrachx.openfood.network.deserializers.** { *; }
 -ignorewarnings
 #Keep Jackson classes ( https://sourceforge.net/p/proguard/discussion/182456/thread/e4d73acf/ )
--keepnames class org.codehaus.jackson.** {
-    *;
-}
+-keepnames class org.codehaus.jackson.** { *; }
+-keepnames class com.fasterxml.jackson.** { *; }
+-keepnames interface com.fasterxml.jackson.** { *; }
 

--- a/app/proguard-rules.pro
+++ b/app/proguard-rules.pro
@@ -21,9 +21,15 @@
 -keep class **$$JsonObjectMapper { *; }
 -keep class openfoodfacts.github.scrachx.openfood.models.** { *; }
 -keep class openfoodfacts.github.scrachx.openfood.network.deserializers.** { *; }
--ignorewarnings
+
 #Keep Jackson classes ( https://sourceforge.net/p/proguard/discussion/182456/thread/e4d73acf/ )
 -keepnames class org.codehaus.jackson.** { *; }
 -keepnames class com.fasterxml.jackson.** { *; }
 -keepnames interface com.fasterxml.jackson.** { *; }
 
+#Keep Android Support Library
+-keep public class android.support.v7.widget.** { *; }
+-keep public class android.support.v7.internal.widget.** { *; }
+-keep public class android.support.v7.internal.view.menu.** { *; }
+
+-ignorewarnings

